lib/versioned/^4.0.0/index.js

```diff
@@ -77,16 +71,20 @@ function execute(env, cfg, opts) {
       return fs.writeFileSync(opts.tasksJson, output, 'utf-8');
     }
     try {
-      log.info('Using gulpfile', chalk.magenta(tildify(env.configPath)));
+      log.info(messages.GULPFILE, { path: env.configPath });
       var runMethod = opts.series ? 'series' : 'parallel';
       gulpInst[runMethod](toRun)(function(err) {
         if (err) {
           exit(1);
         }
       });
     } catch (err) {
-      log.error(chalk.red(err.message));
-      log.error('To list available tasks, try running: gulp --tasks');
+      var task = checkTaskNotFound(err);
```

Comment:
@sttk do you want to pull this into a separate PR?

Reply:
Yes. Ideally, it would be better to modify `undertaker` to set these task names as properties of the error to avoid reverse parsing from the error message. Can I modify `undertaker` for this purpose?

Reply:
We will need this transformation for gulp v4 even after we implement the change in undertaker for gulp v5

Reply:
I understood. I'll create a new PR of this `checkTaskNotFound` function for gulp v4.

Reply:
Thank you. When will you have time to submit this?

Reply:
I've create the PR for this. → #262

Reply:
Thank you! I've merged it into this branch